streamingcli/project/deploy_command.py

```diff
@@ -47,93 +53,45 @@ def deploy_project(
         project_name = docker_image_repository or local_project_config.project_name
         profile_data = ProfileAdapter.update_profile_data(
             profile_data=profile_data,
+            deployment_mode=deployment_mode,
             ververica_url=ververica_url,
             ververica_namespace=ververica_namespace,
             ververica_deployment_target_name=ververica_deployment_target_name,
             ververica_webtoken_secret=ververica_webtoken_secret,
             docker_registry_url=docker_registry_url,
+            k8s_namespace=k8s_namespace,
         )
-        ProjectDeployer.validate_profile_data(
-            profile_data=profile_data, docker_image_tag=docker_image_tag
+        deployment_adapter = DeploymentAdapterFactory.get_adapter(
+            profile_data.deployment_mode,
+            profile_data=profile_data,
+            docker_image_tag=docker_image_tag,
+            project_name=project_name,
         )
 
+        deployment_adapter.validate_profile_data()
```

Comment:
Shouldn't validation happen "automatically" (e.g. right in the constructor of the DeploymentAdapter?)